tests/phpunit/tests/html-api/wpHtmlProcessorSemanticRulesHeadingElements.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Unit tests covering WP_HTML_Processor compliance with HTML5 semantic parsing rules
+ * for the H1 - H6 heading elements.
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ *
+ * @since 6.5.0
+ *
+ * @group html-api
+ *
+ * @coversDefaultClass WP_HTML_Processor
+ */
+class Tests_HtmlApi_WpHtmlProcessorSemanticRulesHeadingElements extends WP_UnitTestCase {
+	/*******************************************************************
+	 * RULES FOR "IN BODY" MODE
+	 *******************************************************************/
+
+	/**
+	 * Verifies that H1 through H6 elements generate implied end tags.
+	 *
+	 * @ticket 60060
+	 *
+	 * @covers WP_HTML_Processor::step
```

Comment:
Isn't it `next_tag` we're using here?

Reply:
more specifically this is covering the semantic rules inside `step` 🤷‍♂️